repository/src/main/java/org/alfresco/repo/security/authentication/ResetPasswordServiceImpl.java

```diff
@@ -373,7 +373,7 @@ public void sendResetPasswordEmail(DelegateExecution execution, String fallbackE
 
         final ClientApp clientApp = getClientAppConfig(clientName);
         Map<String, Serializable> emailTemplateModel = Collections.singletonMap(FTL_RESET_PASSWORD_URL,
-                    createResetPasswordUrl(clientApp, id, key));
+                    createResetPasswordUrl(clientApp, id, key, userName));
```

Comment:
Why you need to pass the userName? The username is defined in the workflow properties. 
Probably I'm missing something here, but how this method `sendResetPasswordEmail` supposed to be called? this method will not work if you don't start the workflow first (via `requestReset` method), and there you'll see that the username is added to the workflow properties.

So, based on the ACA-4619 ticket use cases:
- As a user i would like to change my password
  - I don't think this case can be covered by this service. As I mentioned you need to start the workflow first...
- As a user i would like to request to change my password at login  'forgot my password' 
  - In this case, the username is already provided and there is no need to add the username explicitly.